examples/utils/example-utils/src/migrator/migrator.ts

```diff
@@ -124,23 +124,40 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
 			// appropriate action (see then() block below).
 
 			const prepareTheMigration = async () => {
-				// It's possible that our modelLoader is older and doesn't understand the new acceptedVersion.
+				// It's possible that our modelLoader is older and doesn't understand the new acceptedMigration.
 				// Currently this fails the migration gracefully and emits an event so the app developer can know
 				// they're stuck. Ideally the app developer would find a way to acquire a new ModelLoader and move
 				// forward, or at least advise the end user to refresh the page or something.
 				// TODO: Does the app developer have everything they need to dispose gracefully when recovering with
 				// a new ModelLoader?
-				const migrationSupported = await this.modelLoader.supportsVersion(acceptedVersion);
+				const migrationSupported = await this.modelLoader.supportsVersion(
+					acceptedMigration.newVersion,
+				);
 				if (!migrationSupported) {
-					this.emit("migrationNotSupported", acceptedVersion);
+					this.emit("migrationNotSupported", acceptedMigration.newVersion);
 					this._migrationP = undefined;
 					return;
 				}
 
-				const detachedModel = await this.modelLoader.createDetached(acceptedVersion);
+				const detachedModel = await this.modelLoader.createDetached(
+					acceptedMigration.newVersion,
+				);
 				const migratedModel = detachedModel.model;
 
-				const exportedData = await migratable.exportData();
+				// Here we load the model at the specified sequence number for export.  This way we can ensure we don't include
+				// any local un-ack'd changes or even remote changes that came in too-late.
+				// TODO:  There is risk that a summary comes in after accepting the migration, which will prevent us from loading
+				// the desired sequence number (as the summary will be too-new).  To avoid this, we'd probably need one of the following:
```

Comment:
Do we know what the long-term plans are here? I doubt we have it but it would be nice to link documentation on these options, and maybe do one of the 4 options listed for demonstration.

Reply:
I think this point becomes a planning item as none of these sound super feasible and/or attractive at first glance.  I think (1) and/or (3) is my preference, but needs more design and thought.  Agree that it should have documentation/example ultimately though.